pandapipes/std_types/std_type_class.py

```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
         return obj
 
 
-class PumpStdType(StdType):
+class InterpolationStdType(StdType):
 
-    def __init__(self, name, reg_par):
+    def __init__(self, name, component, int_fct):
         """
 
-        :param name: Name of the pump object
+        :param name: Name of the interpolation object
         :type name: str
-        :param reg_par: If the parameters of a regression function are already determined they \
-                can be directly be set by initializing a pump object
-        :type reg_par: List of floats
+        :param path: Path where the CSV file, defining a interpolation object, is stored
+        :type path: str
+        :return: An object of the interpolation standard type class
+        :rtype: InterpolationStdType
+        """
+        super(InterpolationStdType, self).__init__(name, component)
+        self.int_fct = int_fct
+        self._x_list = None
+        self._y_list = None
+        self._fill_value = 'extrapolate'
+
+    def update_std_type(self, x_list, y_list, fill_value='extrapolate'):
+        int_fct = interpolation_function(x_list, y_list, fill_value)
+        self.int_fct = int_fct
+        self._x_list = x_list
+        self._y_list = y_list
+        self._fill_value = fill_value
+
+    @classmethod
+    def init_std_type(cls, int_st, x_values, y_values, fill_value='extrapolate'):
+        int_st._x_list = x_values
+        int_st._y_list = y_values
+        int_st._fill_value = fill_value
+
+    @classmethod
+    def from_path(cls, name, path):
+        raise NotImplementedError
+
+    @classmethod
+    def from_list(cls, name, x_values, y_values, fill_value='extrapolate'):
+        raise NotImplementedError
+
+    @classmethod
+    def _from_path(cls, path):
+        data = cls.load_data(path)
+        x_values, y_values = _retrieve_data(data)
+        int_fct = interpolation_function(x_values, y_values)
+        return int_fct, x_values, y_values
+
+    @classmethod
+    def _from_list(cls, x_values, y_values, fill_value='extrapolate'):
+        int_fct = interpolation_function(x_values, y_values, fill_value)
+        return int_fct, x_values, y_values, fill_value
+
+    @classmethod
+    def load_data(cls, path):
+        raise NotImplementedError
+
+
+class RegressionStdType(StdType):
+
+    def __init__(self, name, component, reg_par):
         """
-        super(PumpStdType, self).__init__(name, 'pump')
+
```

Comment:
A basic description of this class would be helpful, also to differentiate it from the InterpolationStdType

Reply:
added it

Reply:
thanks